labs/src/main/scala/org/scalalabs/basic/lab03/OptionExercise.scala

```diff
@@ -25,9 +25,29 @@ object OptionExercise01 {
    * - does not exist: 					"not existing"
    */
   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
+    rooms.get(room) map stateString getOrElse "not existing"
   }
-
+  
+  def stateString(opt: Option[String]):String = {
+    opt map (state => if (state == "locked") "not available" else state) getOrElse "empty"
+  }
+  
+  /* Why is the above considered more idiomatic than
+   
+   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
+     rooms.get(room) match {
+     case None => "not existing"
+     case Some opt =>
```

Comment:
Ah, that's really nice! I didn't realize that you could do nested pattern matching so I had stuff like case Some[None] => "empty", which is much less elegant (though I guess it was a couple lines shorter)